File: php/features.php
<?php
declare(strict_types=1);

// BlackbookWiki SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BlackbookWikiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BlackbookWikiBaseFeature();
            case "test":
                return new BlackbookWikiTestFeature();
            default:
                return new BlackbookWikiBaseFeature();
        }
    }
}

File: py/features.py
# BlackbookWiki SDK feature factory

from feature.base_feature import BlackbookWikiBaseFeature
from feature.test_feature import BlackbookWikiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BlackbookWikiBaseFeature(),
        "test": lambda: BlackbookWikiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
